### PR TITLE
Fix up contextvar propagation

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: python/tests/evaluation/test_evaluation.py [hunk 1]
```diff
@@ -41,7 +41,6 @@ def predict(inputs: dict) -> dict:
         },
         num_repetitions=3,
     )
-    results.wait()
```
Comment (agola11) [inline python/tests/evaluation/test_evaluation.py:44]: why was this removed?

Comment (hinthornw) [reply python/tests/evaluation/test_evaluation.py:44]: Wasn't necessary since we were blocking in the function anyway
